src/content/developers/tutorials/erc20-annotated-code/index.md

```diff
@@ -0,0 +1,987 @@
+    }
+````
+
+The `a.add(b)` function is a safe add. In the unlikely case that `a`+`b`>=`2^256` it does not wrap around
+the way normal addition does.
+
+
+```solidity
+
+    /**
+     * @dev Atomically decreases the allowance granted to `spender` by the caller.
+     *
+     * This is an alternative to {approve} that can be used as a mitigation for
+     * problems described in {IERC20-approve}.
+     *
+     * Emits an {Approval} event indicating the updated allowance.
+     *
+     * Requirements:
+     *
+     * - `spender` cannot be the zero address.
+     * - `spender` must have allowance for the caller of at least
+     * `subtractedValue`.
+     */
+    function decreaseAllowance(address spender, uint256 subtractedValue) public virtual returns (bool) {
+        _approve(_msgSender(), spender, _allowances[_msgSender()][spender].sub(subtractedValue, 
+                "ERC20: decreased allowance below zero"));
+        return true;
+    }
+```    
+
+### Functions that Modify Token Information  {#functions-that-modify-token-information}
+
+These are the four functions that do the actual work: `_transfer`, `_mint`, `_burn`, and `_approve`.
+
+#### The \_transfer function  {#_transfer}
+```solidity
+    /**
+     * @dev Moves tokens `amount` from `sender` to `recipient`.
+     *
+     * This is internal function is equivalent to {transfer}, and can be used to
+     * e.g. implement automatic token fees, slashing mechanisms, etc.
+     *
+     * Emits a {Transfer} event.
+     *
+     * Requirements:
+     *
+     * - `sender` cannot be the zero address.
+     * - `recipient` cannot be the zero address.
+     * - `sender` must have a balance of at least `amount`.
+     */
+    function _transfer(address sender, address recipient, uint256 amount) internal virtual {
+```
+
+This function, `_transfer`, transfers tokens from one account to another. It is called by both
+`transfer` (for transfers from the sender's own account) and `transferFrom` (for using allowances
+to transfer from somebody else's account).
+
+
+&nbsp;
+
+```solidity
+        require(sender != address(0), "ERC20: transfer from the zero address");
+        require(recipient != address(0), "ERC20: transfer to the zero address");
+```
+
+Nobody actually owns address zero in Ethereum (that is, nobody knows a private key whose matching public key
+is transformed to the zero address). When people use that address, it is usually a software bug - so we 
+fail if the zero address is used as the sender or the recipient.
+
+
+&nbsp;
+
+```solidity
+        _beforeTokenTransfer(sender, recipient, amount);
+        
+```
+
+
+There are two ways to use this contract:
+
+1. Use it as a template for your own code
+1. [Inherit from it](https://www.bitdegree.org/learn/solidity-inheritance), and override only those functions that you need to modify
+
+The second method is much better because the OpenWhisk ERC-20 code has already been audited and shown to be secure. When you use inheritence
```

Comment:
```suggestion
The second method is much better because the OpenZeppelin ERC-20 code has already been audited and shown to be secure. When you use inheritance
```